Packages/com.chisel.editor/Chisel/Editor/Editor/Scene/CSGSceneBottomGUI.cs

```diff
@@ -13,670 +13,98 @@ namespace Chisel.Editors
-		static void AxisLockingGUI(ref Rect position, bool showLockLabel)
-		{
-			Space(ref position, kButtonSpace);
-
-			//
-			// Axis locking
-			//
-			if (showLockLabel) Label(ref position, lockContent);
-			CSGEditorSettings.AxisLockX = Toggle(ref position, CSGEditorSettings.AxisLockX, lockXAxisContent, lockButtonToggleColor, lockXButtonStyle);
-			CSGEditorSettings.AxisLockY = Toggle(ref position, CSGEditorSettings.AxisLockY, lockYAxisContent, lockButtonToggleColor, lockYButtonStyle);
-			CSGEditorSettings.AxisLockZ = Toggle(ref position, CSGEditorSettings.AxisLockZ, lockZAxisContent, lockButtonToggleColor, lockZButtonStyle);
-		}
-
-		static float AxisLockingWidth(bool showLockLabel)
-		{
-			return kButtonSpace +
-					(showLockLabel ? StyleWidth(miniLabel, lockContent) : 0) +
-					StyleWidth(lockXButtonStyle, lockXAxisContent) +
-					StyleWidth(lockYButtonStyle, lockYAxisContent) +
-					StyleWidth(lockZButtonStyle, lockZAxisContent);
-		}
-		#endregion
-
-		#region ViewOptions
-		static void ViewOptionsGUI(ref Rect position)
-		{
-			Space(ref position, kButtonSpace);
-
-			//
-			// View Options
-			//
-			MaskField(ref position, viewContent, CSGEditorSettings.ViewOptions, ChangeViewOptions, false);
-		}
-
-		static float ViewOptionsWidth()
-		{
-			return kButtonSpace +
-					StyleWidth(EditorStyles.toolbarDropDown, viewContent);
-		}
-		#endregion
-
-		static void DrawLeftButtons(Rect position, SceneViewBarState state)
-		{
-			directionForward = true;
-
-			position.x--;
-
-			ViewOptionsGUI(ref position);
-			if (!state.hideLocking) AxisLockingGUI(ref position, state.showLockLabel);
-			if (!state.hideMove) MoveSnappingGUI(ref position, state.largeLabels, state.showUnits, state.showIncDec, state.showAxiLabels, state.forceOneAxis);
-			if (!state.hideRotate) RotateSnappingGUI(ref position, state.largeLabels, state.showUnits, state.showIncDec);
-			if (!state.hideScale) ScaleSnappingGUI(ref position, state.largeLabels, state.showUnits, state.showIncDec);
-		}
-
-		static float CalcLeftButtonsWidth(SceneViewBarState state)
-		{
-			return ViewOptionsWidth() +
-				   (!state.hideLocking ? AxisLockingWidth(state.showLockLabel) : 0) +
-				   (!state.hideMove ? MoveSnappingWidth(state.largeLabels, state.showUnits, state.showIncDec, state.showAxiLabels, state.forceOneAxis) : 0) +
-				   (!state.hideRotate ? RotateSnappingWidth(state.largeLabels, state.showUnits, state.showIncDec) : 0) +
-				   (!state.hideScale ? ScaleSnappingWidth(state.largeLabels, state.showUnits, state.showIncDec) : 0);
-		}
-
-		static float CalcRightButtonsWidth()
-		{
-			return kButtonSpace +
-				   StyleWidth(EditorStyles.toolbarButton, rebuildContent);
-		}
-
-		static void DrawRightButtons(Rect position)
-		{
-			directionForward = false;
-
-			position.x = position.width;
-
-			Space(ref position, kButtonSpace);
-
-			//
-			// Rebuild all
-			//
-			if (Button(ref position, rebuildContent))
-			{
+			if (GUILayout.Button("Rebuild", buttonStyle)) { 
```

Comment:
I would prefer if strings like this are stored in static GUIContents, it avoid allocations and allows us to specifiy a tooltip as well

Reply:
Will this button ever actually be useful? I kept it in but i'm never in a situation where it needs to be rebuilt

Reply:
Well it was a failsafe in realtime-csg in case a bug popped up. It resets all the internal caches and rebuilds everything. It helped people a lot over the years. Of course realtime csg had some design issues and Chisel *should* be more solid..

Reply:
Yeah I guess one concern is that having a rebuild button, especially always visible on the toolbar, implies to the user that it is a necessary thing. We could maybe hide the button in a menu and give it a hotkey just so it doesn't come across as a necessary part of use, but still available just in case.

Reply:
Well that's why in realtime-CSG i put it all the way to the right and had a simple small icon for it. Thing is that if you run into issues regularly you want to be convenient

Reply:
Right yeah, I think having a hotkey for it can alleviate that in the case that a user needs to use the button often (which is hopefully never). It's not part of the excepted or wanted user flow for them to be using it very often, but there's always the possibility it could happen, so I feel a hotkey is a good balance here.